indico/modules/rb/operations/bookings.py

```diff
@@ -349,8 +349,10 @@ def create_booking_for_event(room_id, event):
         return None
 
 
-def get_active_bookings(limit, start_dt, last_reservation_id=None, **filters):
+def get_active_bookings(limit, start_dt, end_dt=None, last_reservation_id=None, **filters):
     criteria = [ReservationOccurrence.start_dt > start_dt]
+    if end_dt is not None:
+        criteria[0] = (db.and_(criteria[0], ReservationOccurrence.end_dt <= end_dt))
```

Comment:
IMHO this would be more readable:

```py
    if end_dt:
        criteria = [(ReservationOccurrence.start_dt > start_dt) & (ReservationOccurrence.end_dt <= end_dt)]
    else:
        criteria = [ReservationOccurrence.start_dt > start_dt]
```

PS: I think using `>=` instead of `>` here would remove the need for the 1 second subtraction in the frontend code. But ignore this if it's needed for the endless scrolling.

Reply:
> PS: I think using >= instead of > here would remove the need for the 1 second subtraction in the frontend code. But ignore this if it's needed for the endless scrolling.

Yes, the `>` operator is used when there is a `last_reservation_id`, and apparently it's not easy to change.

Reply:
Done in [64f55e6](https://github.com/indico/indico/pull/6846/commits/64f55e6aeae81313fa83df2919cffb514d51459c).